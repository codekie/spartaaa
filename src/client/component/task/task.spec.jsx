import React from 'react';
import { shallow } from 'enzyme';
import Task from './task.jsx';

const TIMESTAMP__DUE = 1541687666487;

it('renders correctly', () => {
    let _origNow = Date.now;

    beforeEach(() => {
        Date.now = jest.fn(() => TIMESTAMP__DUE + 1000);
    });
    afterEach(() => {
        Date.now = _origNow;
    });

    expect(shallow(
        <Task deactivateTask={jest.fn()} activateTask={jest.fn()} finishTask={jest.fn()} id={1}
            unfinishTask={jest.fn()} uuid="ABC" project="spartaaa" due={TIMESTAMP__DUE} />)
    ).toMatchSnapshot();
});
