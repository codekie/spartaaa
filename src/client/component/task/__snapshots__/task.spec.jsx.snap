// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Media
  className="cmp-task undefined"
  renderAs="article"
  style={Object {}}
>
  <div
    className="priority "
  />
  <MediaItem
    className=""
    position="left"
    renderAs="figure"
    style={Object {}}
  >
    <div>
      <Icon
        align={null}
        className="is-medium fa-2x"
        color={null}
        icon={null}
        size={null}
        style={Object {}}
      >
        <FontAwesomeIcon
          border={false}
          className="task-icon"
          fixedWidth={false}
          flip={null}
          icon={null}
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </Icon>
    </div>
    <div>
      <Button
        className="btn-check"
        color={null}
        disabled={false}
        fullwidth={false}
        inverted={false}
        isStatic={false}
        loading={false}
        onClick={[Function]}
        outlined={false}
        remove={false}
        renderAs="button"
        reset={false}
        rounded={false}
        size={null}
        state={null}
        style={Object {}}
        submit={false}
        text={false}
      >
        <Icon
          align={null}
          className=""
          color={null}
          icon={null}
          size={null}
          style={Object {}}
        >
          <FontAwesomeIcon
            border={false}
            className="check-icon"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f00c",
                  "M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z",
                ],
                "iconName": "check",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </Icon>
      </Button>
    </div>
  </MediaItem>
  <MediaItem
    className=""
    position="center"
    renderAs="div"
    style={Object {}}
  >
    <Content
      renderAs="div"
      size={null}
      style={Object {}}
    >
      <p
        className="sub-description"
      >
        <strong
          className="sub-title"
        >
          [
          1
          ] 
        </strong>
      </p>
      <Level
        breakpoint={null}
        className=""
        renderAs="div"
        style={Object {}}
      >
        <LevelSide
          align="left"
          className="tags"
          renderAs="div"
          style={Object {}}
        >
          <LevelItem
            className=""
            renderAs="div"
            style={Object {}}
          >
            <TagGroup
              className="tag-due tag-icon"
              gapless={true}
              style={Object {}}
            >
              <Tag
                className=""
                color="dark"
                remove={false}
                renderAs="span"
                rounded={false}
                size={null}
                style={Object {}}
              >
                <Icon
                  align={null}
                  className="is-small"
                  color={null}
                  icon={null}
                  size={null}
                  style={Object {}}
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          512,
                          512,
                          Array [],
                          "f017",
                          "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z",
                        ],
                        "iconName": "clock",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />
                </Icon>
              </Tag>
              <Tag
                className="overdue"
                color={null}
                remove={false}
                renderAs="span"
                rounded={false}
                size={null}
                style={Object {}}
              >
                an hour ago
              </Tag>
            </TagGroup>
          </LevelItem>
          <LevelItem
            className=""
            renderAs="div"
            style={Object {}}
          >
            <TagGroup
              className="tag-project tag-icon"
              gapless={true}
              style={Object {}}
            >
              <Tag
                className=""
                color="dark"
                remove={false}
                renderAs="span"
                rounded={false}
                size={null}
                style={Object {}}
              >
                <Icon
                  align={null}
                  className="is-small"
                  color={null}
                  icon={null}
                  size={null}
                  style={Object {}}
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          512,
                          512,
                          Array [],
                          "f07b",
                          "M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z",
                        ],
                        "iconName": "folder",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />
                </Icon>
              </Tag>
              <Tag
                className=""
                color="info"
                remove={false}
                renderAs="span"
                rounded={false}
                size={null}
                style={Object {}}
              >
                spartaaa
              </Tag>
            </TagGroup>
          </LevelItem>
          <LevelItem
            className=""
            renderAs="div"
            style={Object {}}
          >
            <Connect(Component)
              uuid="ABC"
            />
          </LevelItem>
        </LevelSide>
      </Level>
      <Level
        breakpoint={null}
        className="aux-bar-1"
        renderAs="div"
        style={Object {}}
      >
        <LevelSide
          align="left"
          className=""
          renderAs="div"
          style={Object {}}
        >
          <Task
            activateTask={[Function]}
            deactivateTask={[Function]}
            isActive={false}
            isCompleted={false}
            taskId={1}
          />
        </LevelSide>
      </Level>
    </Content>
    <Level
      breakpoint="mobile"
      className=""
      renderAs="div"
      style={Object {}}
    >
      <LevelSide
        align="left"
        className=""
        renderAs="div"
        style={Object {}}
      />
    </Level>
  </MediaItem>
  <MediaItem
    className=""
    position="right"
    renderAs="div"
    style={Object {}}
  >
    <Content
      renderAs="div"
      size={null}
      style={Object {}}
    >
      <small />
    </Content>
  </MediaItem>
</Media>
`;
